### DISCUSSION TITLE
Dealing with Pre-existing crack

### DISCUSSION THREAD

**Discussion by guangdongzhang** [peridigm/peridigm#292] [Q&A]
Hello Everyone,
          I wonder how does the Peridigm deal with pre-existing crack, should we just remove the elements in the geomotry before we get the input to peridigm? Is it enough for this process? What happens when the crack thickness is within the horizon size, what's the way does the code deal with the crack, this is the main question that I would like to request. Much appreaciate!

**Comment by oldninja** [comment]
Hello @guangdongzhang,

> should we just remove the elements in the geometry before we get the input to peridigm?

No, you can keep all your elements and insert a `bond filter`, see this example: [Precracked Plate](https://github.com/peridigm/peridigm/blob/master/test/regression/PrecrackedPlate/PrecrackedPlate.xml). There is a discretization parameter, and it is associated with a bond filtering for a rectangular plane.

> What happens when the crack thickness is within the horizon size, what's the way does the code deal with the crack, this is the main question that I would like to request.

If I recall correctly, the bond filter is applied where you place it, and all bond interactions within that region get filtered. Meaning that even if the horizon size is larger than the crack, those interactions are removed, not only those lying on the surface of the crack. See the image below; I inserted a crack and this is how it is processed:

![Number of Neighbors](https://user-images.githubusercontent.com/49734546/212559381-89dc5174-e560-4c88-a6c8-4c7d3ea37447.png)
![Bond Filter](https://user-images.githubusercontent.com/49734546/212559464-32f6fddf-7c2a-4f81-9c65-36ee1a8a7808.png)
